extensions.go

```diff
@@ -567,6 +567,56 @@ func (ed *earlyDataExtension) Unmarshal(data []byte) (int, error) {
 	return 2 + configLen + 2 + extLen + 1 + contextLen, nil
 }
 
+// opaque ProtocolName<1..2^8-1>;
+//
+// struct {
+//     ProtocolName protocol_name_list<2..2^16-1>
+// } ProtocolNameList;
+type alpnExtension struct {
+	protocols []string
+}
+
+func (alpn alpnExtension) Type() helloExtensionType {
+	return extensionTypeALPN
+}
+
+func (alpn alpnExtension) Marshal() ([]byte, error) {
+	listData := []byte{}
+	for _, proto := range alpn.protocols {
+		listData = append(listData, byte(len(proto)))
+		listData = append(listData, []byte(proto)...)
```

Comment:
That's how you append another slice, vs. appending a single element (as on the previous line).